This means static linkage requires -DID3LIB_LINKOPTION=LINKOPTION_STATIC

--- origsrc/id3lib-3.8.3/include/id3/globals.h	2003-03-01 18:23:00.000000000 -0600
+++ src/id3lib-3.8.3/include/id3/globals.h	2013-12-05 16:48:20.247201100 -0600
@@ -46,34 +46,36 @@
 #  define LINKOPTION_CREATE_DYNAMIC 2 //should only be used by prj/id3lib.dsp
 #  define LINKOPTION_USE_DYNAMIC    3 //if your project links id3lib dynamic
 #  ifndef ID3LIB_LINKOPTION
-#    pragma message("*** NOTICE *** (not a real error)")
-#    pragma message("* You should include a define in your project which reflect how you link the library")
-#    pragma message("* If you use id3lib.lib or libprj/id3lib.dsp (you link static) you should add")
-#    pragma message("* ID3LIB_LINKOPTION=1 to your preprocessor definitions of your project.")
-#    pragma message("* If you use id3lib.dll (you link dynamic) you should add ID3LIB_LINKOPTION=3")
-#    pragma message("* to your preprocessor definitions of your project.")
-#    pragma message("***")
-#    error read message above or win32.readme.first.txt
-#  else
+#    ifdef INSIDE_ID3LIB
+#      ifdef PIC
+#        define ID3LIB_LINKOPTION LINKOPTION_CREATE_DYNAMIC
+#      else
+#        define ID3LIB_LINKOPTION LINKOPTION_STATIC
+#      endif
+#    else
+#      define ID3LIB_LINKOPTION LINKOPTION_USE_DYNAMIC
+#    endif
+#  endif
 #    if (ID3LIB_LINKOPTION == LINKOPTION_CREATE_DYNAMIC)
        //used for creating a dynamic dll
-#      define ID3_C_EXPORT extern _declspec(dllexport)
+#      define ID3_C_EXPORT extern __declspec(dllexport)
 #      define ID3_CPP_EXPORT __declspec(dllexport)
-#      define CCONV __stdcall // Added for VB & Delphi Compatibility - By FrogPrince Advised By Lothar
 #    endif
 #    if (ID3LIB_LINKOPTION == LINKOPTION_STATIC)
        //used for creating a static lib and using a static lib
 #      define ID3_C_EXPORT
 #      define ID3_CPP_EXPORT
-#      define CCONV
 #    endif
 #    if (ID3LIB_LINKOPTION == LINKOPTION_USE_DYNAMIC)
        //used for those that do not link static and are using the dynamic dll by including a id3lib header
-#      define ID3_C_EXPORT extern _declspec(dllimport)
+#      define ID3_C_EXPORT extern __declspec(dllimport)
 #      define ID3_CPP_EXPORT __declspec(dllimport) //functions like these shouldn't be used by vb and delphi,
+#    endif
+#    if defined(_WIN64) || defined(LINKOPTION_STATIC)
+#      define CCONV
+#    else
 #      define CCONV __stdcall // Added for VB & Delphi Compatibility - By FrogPrince Advised By Lothar
 #    endif
-#  endif
 #else /* !WIN32 */
 #  define ID3_C_EXPORT
 #  define ID3_CPP_EXPORT
--- origsrc/id3lib-3.8.3/src/Makefile.am	2013-12-05 15:31:46.279441200 -0600
+++ src/id3lib-3.8.3/src/Makefile.am	2013-12-05 16:14:33.394271800 -0600
@@ -18,6 +18,7 @@ zlib_include =
 endif
 
 INCLUDES = \
+  -DINSIDE_ID3LIB \
   @ID3LIB_DEBUG_FLAGS@ -I$(top_srcdir)/include/id3 -I$(top_srcdir)/include $(zlib_include)
 
 noinst_HEADERS =                \
--- origsrc/id3lib-3.8.3/src/globals.cpp	2003-03-01 18:23:00.000000000 -0600
+++ src/id3lib-3.8.3/src/globals.cpp	2013-12-05 16:49:53.234519700 -0600
@@ -35,14 +35,14 @@ extern "C"
 {
 #endif
 
-  const char * const ID3LIB_NAME           = _ID3LIB_NAME;
-  const char * const ID3LIB_VERSION        = _ID3LIB_VERSION;
-  const char * const ID3LIB_FULL_NAME      = _ID3LIB_FULLNAME;
-  const int          ID3LIB_MAJOR_VERSION  = _ID3LIB_MAJOR_VERSION;
-  const int          ID3LIB_MINOR_VERSION  = _ID3LIB_MINOR_VERSION;
-  const int          ID3LIB_PATCH_VERSION  = _ID3LIB_PATCH_VERSION;
-  const int          ID3LIB_INTERFACE_AGE  = _ID3LIB_INTERFACE_AGE;
-  const int          ID3LIB_BINARY_AGE     = _ID3LIB_BINARY_AGE;
+ID3_C_EXPORT const char * const ID3LIB_NAME           = _ID3LIB_NAME;
+ID3_C_EXPORT const char * const ID3LIB_VERSION        = _ID3LIB_VERSION;
+ID3_C_EXPORT const char * const ID3LIB_FULL_NAME      = _ID3LIB_FULLNAME;
+ID3_C_EXPORT const int          ID3LIB_MAJOR_VERSION  = _ID3LIB_MAJOR_VERSION;
+ID3_C_EXPORT const int          ID3LIB_MINOR_VERSION  = _ID3LIB_MINOR_VERSION;
+ID3_C_EXPORT const int          ID3LIB_PATCH_VERSION  = _ID3LIB_PATCH_VERSION;
+ID3_C_EXPORT const int          ID3LIB_INTERFACE_AGE  = _ID3LIB_INTERFACE_AGE;
+ID3_C_EXPORT const int          ID3LIB_BINARY_AGE     = _ID3LIB_BINARY_AGE;
 
 #ifdef _cplusplus
 }
